### PR TITLE
Add basic support for yubikey otp 2fa

### PR DESCRIPTION
**Pull request by hdiniz**
Related https://github.com/opencollective/opencollective/issues/5294

Add yubikey otp 2fa provider and device id column to Users table.

This will be used as MVP for internal users only, device ids will be inserted directly at this time.

### CODE REVIEW

**Comment by hdiniz** [comment]
> Small thing, we should use kebab case for files. https://github.com/opencollective/opencollective/blob/main/rfcs/004-file-naming-convention.md#2-file-names-should-be-kebab-case
> 
>     * recovery_code.ts -> recovery-code.ts
> 
>     * yubikey_otp.ts -> yubikey-otp.ts

Thanks, done!

File: migrations/20230309234219-add-yubikeyDeviceId-to-user-table.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+'use strict';
+
+/** @type {import('sequelize-cli').Migration} */
+module.exports = {
+  async up(queryInterface, Sequelize) {
+    await queryInterface.addColumn('Users', 'yubikeyDeviceId', {
+      type: Sequelize.STRING,
+      allowNull: true,
+      defaultValue: null,
+      unique: true,
```
Comment (Betree) [inline migrations/20230309234219-add-yubikeyDeviceId-to-user-table.js:10]: I don't think we should make that one unique, it's a legit use case to share the same key for multiple accounts (e.g. I use the same one for my personal and OC Cloudflare account)

```suggestion
```

File: server/lib/two-factor-authentication/yubikey-otp.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      throw new ApolloError('Two-factor authentication code is invalid', 'INVALID_2FA_CODE');
+    }
+  },
+};
+
+async function validateYubikeyOTP(otp: string): Promise<boolean> {
+  const response = await fetch(
+    `https://api2.yubico.com/wsapi/2.0/verify?${new URLSearchParams({
+      otp,
+      nonce: crypto.randomBytes(16).toString('base64'),
```
Comment (Betree) [inline server/lib/two-factor-authentication/yubikey-otp.ts:35]: What's the purpose of this nonce? Are we suppose to read it back in the response?

**Review by Betree — APPROVED** [review]
The code is simple - as described in the kickoff. Good job!

File: server/lib/two-factor-authentication/yubikey-otp.ts (hunk 2 above)
Comment (hdiniz) [reply server/lib/two-factor-authentication/yubikey-otp.ts:35]: Its a required field on the API, I am not sure if we need to validate it on the response. From what I could gather its used by the API to differentiate between a replayed OTP (otp only) and a replayed request (same OTP and nonce)

**Pull request merged**